pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1548,7 +1575,9 @@ func (n *NetworkPolicyController) deleteInternalNetworkPolicy(name string) {
 	internalNetworkPolicy := obj.(*antreatypes.NetworkPolicy)
 	n.internalNetworkPolicyStore.Delete(internalNetworkPolicy.Name)
 	n.cleanupOrphanGroups(internalNetworkPolicy)
-
+	if features.DefaultFeatureGate.Enabled(features.Multicluster) && internalNetworkPolicy.SourceRef.Type != controlplane.K8sNetworkPolicy {
```

Comment:
Better to pass a new bool parameter for `NewNetworkPolicyController` like `multiclusterEnabled` to check if multi-cluster is enable or not from `cmd/antrea-agent/agent.go` with below assignment:
```
multiclusterEnabled := features.DefaultFeatureGate.Enabled(features.Multicluster) && o.config.Multicluster.Enable
```

Reply:
Done